### DISCUSSION TITLE
Manually update persistence file

### DISCUSSION THREAD

**Discussion by cavfiumella** [python-telegram-bot/python-telegram-bot#2834] [Q&A] [answered]
Hi there,
I am writing a bot that automatically sends reports on Covid with a rate defined by user (i.e. daily, weekly, monthly).
To achieve this I configured some commands to set reports settings and start receiving updates.
Settings are stored in `chat_data` and bot uses `PicklePersistence`.

When a report is sent successfully a new value (i.e. `last_report`) is inserted in `chat_data` to be read at the next sending attempt to avoid multiple submissions of the same message.
Reports are sent by a separate thread that runs continuously and that is not called as a callback but started when bot is started.

My problem is that when a callback is triggered and `chat_data` changes, pickle file is updated at the same time but when my thread add the last report sending time to `chat_data`, pickle file is not updated until a new callback is done.

Is there a way using library API to trigger pickle file update without "manually" writing the change in the pickle file?

Thank you for the help.

**Comment by cavfiumella** [comment, marked as answer]
Probably I found the solution searching the documentation:
`PicklePersistence.flush()` seems not to work in my case but `PicklePersistance.update_chat_data(chat_id, data)` works fine.

**Reply by Bibo-Joshi** [reply]
The interface if Pickle/BasePersitence ist intended to be accessed exclusively by the Dispatcher. I highly recommend to call Dispatcher.update_persistence instead to avoid overriding any data by accident.

**Comment by ghost** [comment]
I want to know what is other choice for picklepersistence

**Reply by Bibo-Joshi** [reply]
`PickePersistence` is the only presistence implementation that PTB provides for out-of-the-box usage. You can however implement your own persistence backend: https://github.com/python-telegram-bot/python-telegram-bot/wiki/Making-your-bot-persistent#3rd-party-persistence-classes
